src/content/docs/en/reference/loader-reference.mdx

```diff
@@ -3,6 +3,450 @@ title: Astro Loader API
+
+Get all the keys of the entries in the collection.
+
+#### `values`
+
+<p>
+
+**Type**: `() => Array<DataEntry>`
+</p>
+
+Get all entries in the collection as an array.
+
+#### `delete`
+
+<p>
+
+**Type**: `(key: string) => void`
+</p>
+
+Delete an entry from the store by its ID.
+
+#### `clear`
+
+<p>
+
+**Type**: `() => void`
+</p>
+
+Clear all entries from the collection.
+
+#### `has`
+
+<p>
+
+**Type**: `(key: string) => boolean`
+</p>
+
+Check if an entry exists in the store by its ID.
+
+### `DataEntry`
+
+This is the type of the object that is stored in the data store. It has the following properties:
+
+#### `id`
+
+<p>
+
+**Type**: `string`
+</p>
+
+An identifier for the entry, which must be unique within the collection. This is used to look up the entry in the store, and is the key used with `getEntry` for that collection. 
+
+#### `data`
+
+<p>
+
+**Type**: `Record<string, unknown>`
+</p>
+
+The actual data for the entry. When a user accesses the collection, this will have TypeScript types generated according to the collection schema. It the responsibility of the loader to use [`parseData`](#parsedata) to validate and parse the data before storing it in the data store: no validation is done when getting or setting the data.
+
+#### `filePath`
+
+<p>
+
+**Type**: `string | undefined`
+</p>
+
+A path to the file that is the source of this entry, relative to the root of the site. This only applies to file-based loaders, and is used to resolve paths such as images or other assets. If not set, then any fields in the schema that use [the `image()` helper](/en/guides/images/#images-in-content-collections) will be treated as [public paths](/en/guides/images/#where-to-store-images) and not transformed.
+
+#### `body`
+
+<p>
+
+**Type**: `string | undefined`
+</p>
+
+The raw body of the entry, if applicable. If the entry includes [rendered content](#rendered), then this field can be used to store the raw source. This is optional, and is not used internally.
+
+#### `digest`
+
+<p>
+
+**Type**: `string | undefined`
+</p>
+
+An optional content digest for the entry. This can be used to check if the data has changed. If the digest is passed when setting an entry, it will only update the entry if the digest does not match the existing entry. The format of the digest is up to the loader, but it should be a string that changes when the data changes. The [`generateDigest`](#generatedigest) function can be used for this.
+
+#### `rendered`
+
+<p>
+
+**Type**: `RenderedContent | undefined`
+</p>
+
+If the entry has been rendered to HTML, this field can be used to store an object with the rendered content and metadata. This can be used to store the rendered content of a markdown entry, for example, or HTML from a CMS. This field is optional, but if provided then the user can use [the `render()` function and `<Content />` component](/en/guides/working-with-collections/#rendering-body-content) to render the entry in a page.
```

Comment:
```suggestion
Stores an object with an entry's rendered content and metadata if it has been rendered to HTML. For example, this can be used to store the rendered content of a Markdown entry or HTML from a CMS.

If this field is provided, then [the `render()` function and `<Content />` component](/en/guides/working-with-collections/#rendering-body-content) are available to render the entry in a page.
```